eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnmodifiableMutableMap.java

```diff
@@ -390,6 +391,12 @@ public boolean equals(Object obj)
         return (MutableMap<K, V>) this.delegate;
     }
 
+    @Override
+    public MutableSet<K> keySet()
+    {
+        return SetAdapter.adapt(this.delegate.keySet()).asUnmodifiable();
```

Comment:
@johnjstuart you can write instead: `getMutableMap().asUnmodifiable()` I believe

Reply:
@johnjstuart sorry, I meant `getMutableMap().keySet().asUnmodifiable()` (since now `MutableMap` has a `keySet()` that returns `MutableSet` I believe).

Reply:
@johnjstuart I agree with @victornoel recommendation here.